src/server/game/Server/Protocol/Opcodes.cpp

```diff
@@ -1396,7 +1396,7 @@ void OpcodeTable::Initialize()
     /*0x4FD*/ DEFINE_SERVER_OPCODE_HANDLER(SMSG_LOOT_SLOT_CHANGED,                                  STATUS_NEVER);
     /*0x4FE*/ DEFINE_HANDLER(UMSG_UPDATE_GROUP_INFO,                                                STATUS_NEVER,      PROCESS_INPLACE,        &WorldSession::Handle_NULL                              );
     /*0x4FF*/ DEFINE_HANDLER(CMSG_READY_FOR_ACCOUNT_DATA_TIMES,                                     STATUS_AUTHED,     PROCESS_THREADUNSAFE,   &WorldSession::HandleReadyForAccountDataTimes           );
-    /*0x500*/ DEFINE_HANDLER(CMSG_QUERY_QUESTS_COMPLETED,                                           STATUS_LOGGEDIN,   PROCESS_THREADSAFE,     &WorldSession::HandleQueryQuestsCompleted               );
+    /*0x500*/ DEFINE_HANDLER(CMSG_QUERY_QUESTS_COMPLETED,                                           STATUS_LOGGEDIN,   PROCESS_INPLACE,        &WorldSession::HandleQueryQuestsCompleted               );
```

Comment:
What determines if it should be inplace/threadsafe/threadunsafe?

Reply:
```cpp
enum PacketProcessing
{
    PROCESS_INPLACE = 0,                                    //process packet whenever we receive it - mostly for non-handled or non-implemented packets
    PROCESS_THREADUNSAFE,                                   //packet is not thread-safe - process it in World::UpdateSessions()
    PROCESS_THREADSAFE                                      //packet is thread-safe - process it in Map::Update()
};
```